gobblin-runtime/src/main/java/org/apache/gobblin/runtime/api/FsSpecConsumer.java

```diff
@@ -74,6 +78,8 @@ public FsSpecConsumer(Config config) {
       return null;
     }
 
+    Arrays.sort(fileStatuses, Comparator.comparingLong(FileStatus::getModificationTime));
```

Comment:
add a comment for why you're doing it and what you're expecting (ascending sort?)